refinery_macros/src/util.rs

```diff
@@ -0,0 +1,133 @@
+use std::ffi::OsStr;
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use regex::Regex;
+
+pub(crate) enum MigrationType {
+    Mod,
+    Sql,
+}
+
+impl MigrationType {
+    fn file_match_re(&self) -> Regex {
+        let ext = match self {
+            MigrationType::Mod => "rs",
+            MigrationType::Sql => "sql",
+        };
+        let re_str = format!(r"^V\d+(\.\d+)?__\w+\.{}$", ext);
+        Regex::new(re_str.as_str()).unwrap()
+    }
+}
+
+pub(crate) fn crate_root() -> PathBuf {
+    let crate_root = env::var("CARGO_MANIFEST_DIR")
+        .expect("CARGO_MANIFEST_DIR environment variable not present");
+    PathBuf::from(crate_root).canonicalize().unwrap()
+}
+
+pub(crate) fn file_names(
+    input: impl Iterator<Item = PathBuf>,
+    stem_only: bool,
+) -> impl Iterator<Item = String> {
+    input.filter_map(move |entry| {
+        let file_name = if stem_only {
+            entry.file_stem()
+        } else {
+            entry.file_name()
+        };
+        file_name.and_then(OsStr::to_str).map(String::from)
+    })
+}
+
+pub(crate) fn find_migration_files(
```

Comment:
I originally tried to implement this but unfortunately, `refinery_macros` can't depend on `refinery` (circular dependency). To expose it in the public API, I can move the copy in `refinery_cli` to `refinery` but it would still have to exist separately in `refinery_macros`.

To mitigate the code duplication issue, I tried to simplify `find_migration_files()` as much as possible. All it's doing is filtering the contents of a directory with a regex match:

```rust
let file_paths = fs::read_dir(location)?
    .filter_map(Result::ok)
    .map(|de| de.path())
    // filter by migration file regex
    .filter(
        move |entry| match entry.file_name().and_then(OsStr::to_str) {
            Some(file_name) => re.is_match(file_name),
            None => false,
        },
    );
```